Trunk

pschwartz opened the pull request:
LIBCLOUD-88 - Added Swift storge dirver and tests.

Kami committed:
Add support for the new Amazon region (Tokyo).


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1076474 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add missing fixtures.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1079005 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Delete all the files (merging storage_api branch from the git repo).


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1079023 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Merge storage_api branch.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1079029 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
secrets.py shouldn't be here.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1079031 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update version number.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1079032 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changelog.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1079035 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add test for upload_object throwing an Exception.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1079332 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Oops, this whole body shouldn't be wrapped inside "if calculate_hash"


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1079334 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix indentation.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1079335 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Add missing packages to setup.py after refactoring.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1079909 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update paths in providers.py


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1080281 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Rename stream_object_data to upload_object_via_stream

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081240 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Rename object_as_stream to download_object_as_stream

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081242 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add coverage command.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081243 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Start refactoring out common rackspace classes from both cloudfiles and cloud servers

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081244 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Version the auth API separately from the normal APIs

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081247 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add pricing module.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081248 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
remove unused imports

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081249 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
1. Add docstrings to the pricing module
2. Move utility methods from the cloudsigma driver to the utils module


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081404 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update paths


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081405 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add new and fix existing pricing tests,


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081410 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update pricing module and add get_size_price method.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081423 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a typo.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081424 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Always cast size price to float


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081426 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Start updating the drivers to use the pricing module (w.i.p.)


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081493 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Modify more drivers to use the pricing module.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081583 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Include pricing data to package_data.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081819 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
For now also use the pricing module to store the vps.net prices.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081830 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Oops, fix a typo.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081831 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add "medium" size to the elastichosts driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081839 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add pricing for the ElasticHosts driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081845 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Use _get_size_price method in the rackspace driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081849 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Move path stuff into a separate function.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081852 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add set_pricing function.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081870 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Apply Tavis Rodd's patch (style fixes, cloudfiles test fixes, cloudfiles bug
fix)


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081909 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove duplicated methods.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081911 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix softlayer test on 2.7


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1081915 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update the vps.net driver to be consistent with other drivers (now a returned
price is per day instead of per month).


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082010 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
iAdd in_development_warning function.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082027 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add some basic tests for backward compatibility.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082028 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add more tests for Rackspace driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082053 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add tests for deployment.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082055 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add another test for deployment.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082056 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add test for script deployment.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082058 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add tests for some util functions.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082063 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Make mimetype test more flexible.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082065 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Make sure we enter all the branches in deployment tests.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082067 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add more tests for pricing.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082068 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update a list of excluded statements.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082069 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Also run doc tests.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082070 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Ignore more unimportant statements.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082072 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add tests for httplib ssl stuff.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082213 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add another test for SSL stuff.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082241 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove this for now.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082244 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add required line arg.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082245 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add more tests for the Voxel driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082269 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
More tests for Voxel driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082319 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix dummy doctest to work with pypy.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082374 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Don't relay on list ordering.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082377 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Enable DeprecationWarnings


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082379 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Increase GoGrid driver test coverage.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082389 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add test for get_driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082766 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add more tests for ElasticHosts driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082769 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Use constant.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082770 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add Nimbus driver.

Changes submitted by David LaBissoniere <labisso at uchicago dot edu> as a part
of LIBCLOUD-75.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082779 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082781 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add Bluebox compute driver.

Changes submitted by Christian Paredes <cp at redbluemagenta dot com> as a part
of LIBCLOUD-73.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082793 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update CHANGES.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082794 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Forgot to commit this files.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1082805 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add missing variables for brightbox driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1084176 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix style.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1084177 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add extension method for modifying instance attributes and change the node size.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1084180 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1084183 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add docstrings.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1084201 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add Gandi.net compute driver.

Changes submitted by Aymeric Barantal <mric at gandi dot net> as a part
of LIBCLOUD-76.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1085099 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1085100 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add more tests for the Gandi.net compute driver.

Changes submitted by Aymeric Barantal <mric at gandi dot net> as a part
of LIBCLOUD-76.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1085375 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add two new methods to the storage API:
- get_container_cdn_url
- get_object_cdn_url


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1088455 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Move xml utility functions into libcloud.utils


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1089830 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Oops, self shouldn't be here.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1089832 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update file affected by the previous commit.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1089834 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add module with aws utility functions

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1089958 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Modify EC2 driver so now the EC2 response inherits from AWSBaseResponse class.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1089962 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Start working on Amazon S3 driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1089998 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add methods for creating and deleting a container.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090073 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add get_container and get_object method to the S3 driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090221 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add delete_object method to the s3 driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090238 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Escape special characters in the object name.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090241 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Create "Name" tag after creating an EC2 node.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090495 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update CHANGES.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090496 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Make sure both the numbers are integers.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090579 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Move _get_object method to the base StorageDriver class.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090582 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Modify CloudFiles driver to use _get_object method from the base class.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090583 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add download_object method to the S3 driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090584 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix the callback argument.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090587 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix download_objecT_as_stream method in the CloudFiles driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090595 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add download_object_as_stream method to the S3 driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090596 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Move _put_object method from the CloudFiles driver to the base StorageDriver
class, refactor it and make ClouFiles driver use this method.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090613 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
If a provided file_path does not exist, throw an exception early.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090619 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Return RawResponse instance.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090624 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
oops, this shouldn't be here.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090626 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add upload_object method to the S3 driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090628 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix provider constant in the example.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090630 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add explicit notice about S3 and upload_object_via_stream.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090632 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
oops.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090670 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Start working on S3 tests.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090671 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
More tests and fixes for the S3 driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090717 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add new OpSource compute driver.

Changes submitted by Joe Miller <joeym at joeym dot net> as a part
of LIBCLOUD-77.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090725 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a pre connect hook which gets called before sending a request to the remote
server and can perform final manipulation with the requests parameters and
headers.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090850 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
1. Calculate the request signature in the pre_connect hook
2. Pass Content-MD5 header when the file_hash argument is provided


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090851 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Properly handle response when the invalid hash is provided.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090852 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Move some tests and add some new ones.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090862 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add StorageMockHttp class and make CloudFiles mock class use it.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090869 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Refactor MockRawResponse class.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090877 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add more tests for the S3 driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090879 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Refactor more stuff into base MockRawResponse class.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090883 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove some unused stuff.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090886 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Even more tests for the S3 driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090887 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
More tests for the S3 driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090899 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a bug in the S3 driver meta_data header key name.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090900 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Use constants.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090901 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update .ratignore


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090902 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add more stuff to .ratignore


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090904 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Change method name: _clean_name -> _clean_object_name


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090906 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add even more stuff to .ratignore


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090908 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a bug in CloudFiles driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090910 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add more tests for the CloudFiles driver and fix some places where the
base_headers dictionary was manipulated directly without creating a copy first.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1090911 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix provider constant names for the Asia region.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1091021 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
1. Create signature properly if "x-amz-" headers are provided
2. Only send LocationC1.onstraint payload to the server when calling
create_container if the driver's region is not "US Standard"

M    s3.py


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1091023 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Set body to empty string instead of None when location name is not specified.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1091027 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
GoGrid driver: update to API Version 1.8.

Fix offset for recent CHANGES items while I'm here.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1092838 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Fix running specific test example.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1092839 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Import Load Balancers support.

At the moment drivers for two providers are available:
Rackspace and GoGrid. At the moment only basic features
provided by load balancing service are supported, but
they've been tested and should work without problems.

However, there are a lot of work to be done here still,
like adding locations support in a similar fashion as it's
done for compute part, adding blocking mode and refining
implementation in general.

Follow discussion on the maillist to be up-to-date with
these efforts:

http://mail-archives.apache.org/mod_mbox/incubator-libcloud/201104.mbox/<20110404221031.GA26577@reemsky>


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1095180 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Update CHANGES with lb support.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1095182 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Fix test_backward_compatibility test.

It was broken when common gogrid classes were extracted
to libcloud.common.gogrid module.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1095184 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Extend Rackspace driver to allow connection to custom OpenStack installations.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1100118 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Mention OpenStack driver in CHANGES.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1100119 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Do not hardcode secure=False for OpenStack.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1100129 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Fix 0.4.x style 'from libcloud.providers import get_driver' import.

It was broken because 'libcloud.providers' internally tried to
import non-existent stuff from libcloud.compute.providers.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1100233 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a broken constant name, should be CLOUDFILES_UK.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101075 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add missing __init__.py files.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101162 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Throw an exception if an unexpected status code is returned.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101173 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a bug in CF storage driver - update the headers instead of overwriting them.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101197 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add something method stubs which will be added in the future.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101203 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove unused import.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101204 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Start refactoring Rackspace connection class so we will be able to use a
different hostname for each request (needed when acessing cdn stuff in storage
drivers).


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101236 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Allow user to specify a host on per-request basis.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101240 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add implementation for the following methods to the CloudFiles driver:
- enable_container_cdn()
- get_container_cdn_url()
- get_object_cdn_url()


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101255 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Also pass host to the connect method.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101257 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove print statement.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101266 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Forgot to commit this file before.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101267 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101270 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Allow user to specify ssh client timeout for the deploy_node step.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101285 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
oops, also need pass timeout to the parent class here.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101286 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a bug with primary IP address handling in the Slicehost driver. Slicehost
API sometimes returns internal IP as a primary address and we added this IP to
the public_ip instead of private_ip list.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101288 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Better error handling in the GoGrid LB driver - should thrown an exception
instead of returning None.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1101423 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Rename libcloud.resources.lb to libcloud.loadbalancer'


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1102975 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Simply deploy_node a bit and actually retry the process if it fails.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1102979 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
If a ssh timeout is not provided default to 20 seconds.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1102981 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add LBAlgorithm class and set DEFAULT_ALGORITHM variable to ROUND_ROBIN.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1102982 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Rename 'node{,s}' in the load balancer API to be 'member{,s}', as this is less confusing terminology

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1102988 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
remove unused variables

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1102989 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
fix simplejson fallback

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1102990 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
remove unused imports

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1102991 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Have attach_member actually take a LBMember object as its parameter, rather than a raw ip/port

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1102994 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add two new instance variables and utility methods to the base LBDriver class.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103000 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix formatting.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103002 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Add balancer_attach_compute_node, which takes a normal libcloud compute node object, and attempts to add it to the load balancer as a member

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103003 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
LBNode -> LBMember in example_lb.py


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103005 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Rename example_lb.py -> example_loadbalancer.py.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103006 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a typo and make instance variables uppercase.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103009 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add missing attach_member method.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103010 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
1. Don't use kwargs
2. Use the algorithm algorithm and pass it to the api


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103039 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add reverse_dict utility function.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103040 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Don't use kwargs here.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103044 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Pass in the algorithm to the create_balancer method.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103045 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Centralize the new compute drivers list in the CHANGES file

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103047 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Rename the balancer_detail to get_balancer and change the method signature and
return value.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103053 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Rename the balancer_details method to get_balancer and refactor id.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103056 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Do the same with the balancer_detail method in the Rackspace LB driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103058 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Rewrite changes for 0.5, grouping things by relevant driver

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103059 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Sort drivers alphabetically, fixup some formatting

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103061 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Add release date for 0.4.2

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103062 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Update / sync docs with actual args

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103071 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Change the create_balancer method signature and add list_protocols method.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103096 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Remove the LB prefix to most libcloud.loadbalancer.* types

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103098 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
remove unused imports

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103101 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
fix __all__ for load balancers

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103103 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Ooops, used old name here

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103104 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
Remove unused imports

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103105 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
remove unused import

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103108 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update create_balancer method in the GoGrid driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103109 13f79535-47bb-0310-9956-ffa450edef68

pquerna committed:
remove more unused imports (padding my stats yo)

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103110 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update the create_balancer method in rackspace driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103111 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Use DEFAULT_ALGORITHM constant.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103113 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add list_protocols method to the Rackspace LB driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103136 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Forgot to add this file.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103149 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove file_hash and add verify_hash argument and make it consistent across all
the drivers and update affects tests.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103251 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Make get_meta_data and extension method and update affected tests.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103357 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a bug in the gogrid driver and add a test case for it.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103368 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Make sure the correct parameters are sent when creating a load balancer in the
Rackspace driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103371 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add license header and update the example.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103426 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove print statement.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103442 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Properly set object hash value when using get_object method in the CloudFiles
storage driver.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103445 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update affected test.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103446 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Properly capture errors if they happen in the RawResponse part.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103542 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update loadbalancer example.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1103612 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Minor style fixes for loadbalancer example.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1104046 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Remove no longer used resource directory.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1104047 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
Chase rename of LB and LBNode classes for loadbalancers.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1104048 13f79535-47bb-0310-9956-ffa450edef68

novel committed:
GoGrid Compute Driver: add 16GB node size.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1104132 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Also include actual error value when calling str on Error object.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125667 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a bug with using the incorrect URL when performing a raw request.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125687 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Don't lowercase the amz header values in the request signature.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125691 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix Gandi.net tests so they work with Python 2.7

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125693 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Fix a bug with unittest module in Python 2.7 in the GoGrid load-balancer tests.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125695 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a new MockHttpTestCase class which behaves the same as the base MockHttp class, but you can also use assert methods if you class inherits from this one.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125696 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add missing license hader.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125697 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add another missing license header.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125698 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update .ratignore (Ignore loadbalancer fixtures and empty __init__.py files.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125699 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a notice to the balancer_detach_member method.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125748 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Remove type attribute from the Loadbalancer driver classes.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125749 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Rename RACKSPACE load-balancer provider constant to RACKSPACE_US.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125750 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Should be round robin.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125757 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Index needs to start with 1 otherwise Unexpected server error is thrown.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125759 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update gogrid test.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125760 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Revert back to the old behavior and start with 0.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125764 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Should be 'least connect'.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125765 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Capture unexpected server error when creating a new load-balancer and throw a more user-friendly error message.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125772 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a new test for the GoGrid LB driver.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125774 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Forgot to commit this.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125775 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
pyflakes cleanup.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125776 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update the version string.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125795 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Read the version number from the libcloud/__init__.py file

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125801 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Forgot to pop it.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125804 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a note about api refactoring and the compute modules being moved to
libcloud.compute.*.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125809 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update setup.py, include storage packages.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125810 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add missing data file to setup.py.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125812 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update setup.py and MANIFEST.in

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1125819 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
ex_create_tags should be a pass-through in Nimbus driver since Nimbus doesn't support this method.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1126614 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Add a base LibcloudTestCase which eventually all the test cases should inherit from. This class will provide some utility assertion methods.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1126681 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Refactor EC2 test to inherit from LibcloudTestCase and use the new assertion method.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1126684 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Properly handle response errors in the Rackspace driver and only throw InvalidCredsError on 401 status code.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1126805 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes.

git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1126806 13f79535-47bb-0310-9956-ffa450edef68

Kami committed:
Update changes and add missing driver info.


git-svn-id: https://svn.apache.org/repos/asf/incubator/libcloud/trunk@1126969 13f79535-47bb-0310-9956-ffa450edef68

pschwartz committed:
Added openstack base connection class, swift storage driver, and tests for the swift driver.

pschwartz committed:
Fixed provider name in Swift driver class.

Kami commented on libcloud/common/openstack.py:
```diff
@@ -0,0 +1,122 @@
+
+        return value
+
+    def _populate_hosts_and_request_paths(self):
```
This whole function is too much copy & paste from the base Rackspace Connection class.

We should refactor the base Rackspace connection class to allow developer to specify which keys are expected in the auth server response header and use this in the OpenStack / Swift Connection.

pschwartz committed:
Updated to use new base classing for rackspace based off of openstack.

pschwartz committed:
More cleanup.
- added auth_token as static not in auth_headers_keys.
- added a look of auth_headers_keys to determine urls that was missing in OpenstackBaseConnection.

Kami commented:
(just doing some pull request cleanup)

@pschwartz - Those changes have been merged a long time ago, can you please close this pull request?

Thanks!

pschwartz commented:
Closing as it was merged just over 2 years ago.
